addons/mail/models/discuss/mail_guest.py

```diff
@@ -140,3 +139,24 @@ def _format_auth_cookie(self):
         """
         self.ensure_one()
         return f"{self.id}{self._cookie_separator}{self.access_token}"
+
+    def _can_elevate_access(self, access_token, field):
+        if field == "avatar_128":
+            current_partner, current_guest = self.env["res.partner"]._get_current_persona()
+            if current_partner or current_guest:
+                domain = [("channel_member_ids", "any", [("guest_id", "=", self.id)])]
+                if current_partner:
+                    domain.append(
+                        ("channel_member_ids", "any", [("partner_id", "=", current_partner.id)])
+                    )
+                else:
+                    domain.append(
+                        ("channel_member_ids", "any", [("guest_id", "=", current_guest.id)])
+                    )
+                # access to the avatar is allowed if there is a common channel
```

Comment:
While we are at it, I think we can simply these conditions.
There doesn't need to be a common channel, just a channel that is accessible is enough. Other checks are handled by channel ACL.
So the if/else seems overkill, as well as check of current_partner or current_guest then.